Subject: [PATCH] [ams] set the resultTo to null started by a finishing activity

activitya start activityb via startActivityForResult
if activitya is going to finish, activityb's resultTo
should be set null. otherwise, activityb will be put
to the topTask which is not supposed to happen instead of
creating a new taskrecord.

diff --git a/services/core/java/com/android/server/am/ActivityStarter.java b/services/core/java/com/android/server/am/ActivityStarter.java
index bebfc0b..0bc04ed 100644
--- a/services/core/java/com/android/server/am/ActivityStarter.java
+++ b/services/core/java/com/android/server/am/ActivityStarter.java
@@ -1547,6 +1547,9 @@
         }
         mSourceRecord = null;
         mSourceStack = null;
+        if (mStartActivity != null) {
+            mStartActivity.resultTo = null;
+        }
     }
 
     /**
